packages/dds/sequence/src/test/endpointInRangeIndex.spec.ts

```diff
@@ -0,0 +1,228 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { strict as assert } from "assert";
+import { makeRandom } from "@fluid-internal/stochastic-test-utils";
+import { Client } from "@fluidframework/merge-tree";
+import {
+	Interval,
+	IEndpointInRangeIndex,
+	createEndpointInRangeIndex,
+	intervalHelpers,
+} from "../intervalCollection";
+import {
+	assertPlainNumberIntervals,
+	createTestInterval,
+	generateRandomIntervals,
+} from "./intervalIndexUtils";
+
+class TestEndpointInRangeIndex implements IEndpointInRangeIndex<Interval> {
+	private readonly intervals: Interval[];
+
+	constructor() {
+		this.intervals = [];
+	}
+
+	add(interval: Interval) {
+		this.intervals.push(interval);
+	}
+
+	remove(interval: Interval) {
+		const index = this.intervals.findIndex((i) => i === interval);
+		if (index !== -1) {
+			this.intervals.splice(index, 1);
+		}
+	}
+
+	findIntervalsWithEndpointInRange(start: number, end: number): Interval[] {
+		return this.intervals.filter((interval) => interval.end >= start && interval.end <= end);
+	}
+}
+
+describe("findIntervalsWithEndpointInRange", () => {
+	let helpers;
+	let endpointInRangeIndex;
+	let compareFn;
+	let results;
+
+	beforeEach(() => {
+		helpers = intervalHelpers;
+		endpointInRangeIndex = createEndpointInRangeIndex(helpers, undefined as any as Client);
+		// sort the query result by the interval endpoint value
+		compareFn = (a: Interval, b: Interval) => {
+			if (a.end === b.end) {
+				return a.start - b.start;
+			}
+			return a.end - b.end;
+		};
+	});
+
+	describe("finds no intervals", () => {
+		it("when the index is empty", () => {
+			results = endpointInRangeIndex.findIntervalsWithEndpointInRange(1, 1);
+			assert.equal(results.length, 0);
+		});
+
+		beforeEach(() => {
```

Comment:
`beforeEach` blocks apply to all tests in the same suite. So as factored, `"when the index is empty"` isn't actually testing things when the index is empty.

Reply:
seems like we have to move it to a separate test suite, although logically it should belong to the `describe("finds no intervals")` suite. Is there a more optimal approach to address this situation?

Reply:
you could nest another `describe` block for just the cases that want those intervals. Something like this:

```describe("with intervals in the index", () => { it("when the search range doesn't include intervals") }```